test/benchmark/schbench/smp1/bench_results/p50_rps_smp1.json

```diff
@@ -0,0 +1,17 @@
+{
+    "alert": {
+        "threshold": "130%",
+        "bigger_is_better": true
+    },
+    "result_extraction": {
+        "search_pattern": "\\*",
+        "nth_occurrence": 30,
```

Comment:
I have executed the shell locally. We didn't consider such a situation that a benchmark would run several times and generate multiple similar results. Current practice takes the last result as the final result, which wastes other results. Since schbench does not provide an interface for averaging, I think we should do this ourselves. We can process these data in the virtual machine, that is, `run.sh`, or in `bench_linux_and_aster.sh`. I tend to `run.sh`.

```sh
#!/bin/sh

# SPDX-License-Identifier: MPL-2.0

/benchmark/bin/schbench -F 256 -n 5 -r 90 | tee schbench_output.txt

# Extract percentile values marked with * and calculate their averages
awk '
BEGIN {
    # Initialize variables
    sum_wakeup = 0
    count_wakeup = 0
    sum_request = 0
    count_request = 0
    sum_rps = 0
    count_rps = 0
}

# Match Wakeup Latencies section
/Wakeup Latencies percentiles/ {
    in_wakeup = 1
    in_request = 0
    in_rps = 0
}

# Match Request Latencies section
/Request Latencies percentiles/ {
    in_wakeup = 0
    in_request = 1
    in_rps = 0
}

# Match RPS percentiles section
/RPS percentiles/ {
    in_wakeup = 0
    in_request = 0
    in_rps = 1
}

# Extract values marked with *
/\*/ {
    value = $2
    if (in_wakeup) {
        sum_wakeup += value
        count_wakeup++
    } else if (in_request) {
        sum_request += value
        count_request++
    } else if (in_rps) {
        sum_rps += value
        count_rps++
    }
}

# Output the results
END {
    print "Wakeup Latencies percentiles (usec) runtime 90 (s) (528661 total samples)"
    print "* 99.0th: " sum_wakeup / count_wakeup "         (29695 samples)"
    print ""
    print "Request Latencies percentiles (usec) runtime 90 (s) (529274 total samples)"
    print "* 99.0th: " sum_request / count_request "      (46163 samples)"
    print ""
    print "RPS percentiles (requests) runtime 90 (s) (91 total samples)"
    print "* 50.0th: " sum_rps / count_rps "       (31 samples)"
    print ""
    print "average rps: " sum_rps / count_rps
}
' schbench_output.txt
```

Reply:
Thank you for the advice. 

However the schbench does give a accumulated summary at 27/28/29 lines, which means we don't have to calculate by ourselves. (And I don't think the script above is correct here; but that doesn't matter a lot.)

Actually I don't get why a `nth_occurrence` field is bad since it doesn't hurt existing configs and can save us a lot of time writing scripts. Futhermore, it seems like a common and reasonable need to pick one from the matched results when we are doing a pattern matching.

Reply:
Thanks for correcting my misunderstanding. I am OK to add `nth_occurrence` now.